src/renderers/WebGLRenderer.js

```diff
@@ -1434,7 +1425,7 @@ function WebGLRenderer( parameters ) {
 
 	function renderObject( object, scene, camera, geometry, material, group ) {
 
-		object.onBeforeRender( _this, scene, camera, geometry, material, group );
+		object.onBeforeRender( _this, scene, camera, geometry, material, group, _currentViewport );
```

Comment:
I'm not sure this is the best solution. But we need the `_currentViewport` information in `LensFlare`.

Reply:
I think I would add `.getCurrentViewport()` to `WebGLRenderer` instead.

Reply:
I've added `.getViewport()` since the `current` prefix is not used in other methods. I hope it's okay like that...

Reply:
It is tricky, because `viewport` and `currentViewport` are different. `viewport` is the view size the user is setting. `currentViewport` is the viewport being dictated by `ArrayCamera`. So `.getCurrentViewport()` would be more correct.

Maybe your previous approach was better though. However, I didn't like `viewport` being added the last in `onBeforeRender()`. Maybe is not too late yet to change the order?

```js
object.onBeforeRender( _this, scene, camera, _currentViewport, geometry, material, group );
```

As far as I can see, most of the code/examples mainly use `_this, scene, camera`.

Only this example uses `material`:
https://github.com/mrdoob/three.js/blob/dev/examples/webgl_interactive_instances_gpu.html

Reply:
Ah i see. Then let's add `_currentViewport` to the signature and change the mentioned example.